pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -460,24 +461,59 @@ func (metrics *vmiMetrics) updateNetwork(vmStats *stats.DomainStats) {
 	}
 }
 
-func makeVMIsPhasesMap(vmis []*k6tv1.VirtualMachineInstance) map[string]uint64 {
-	phasesMap := make(map[string]uint64)
+type vmiCountMetric struct {
+	phase    string
```

Comment:
To prevent setting this field from the annotation through reflection, but I removed all the reflect code, it's not worth it, so consider this done also :)